OADB/AliPhysicsSelection.cxx

```diff
@@ -251,63 +259,66 @@ UInt_t AliPhysicsSelection::CheckTriggerClass(const AliVEvent* event, const char
   return returnCode;
 }
 
-//______________________________________________________________________________
-Bool_t AliPhysicsSelection::EvaluateTriggerLogic(const AliVEvent* event, AliTriggerAnalysis* triggerAnalysis, const char* triggerLogic, Bool_t offline){
-  // evaluates trigger logic. If called with no event pointer/triggerAnalysis pointer, it just caches the tokens
-  // Fills the statistics histogram, if booked at row i
-  TString trigger(triggerLogic);
-  
-  // add space after each token (to use ReplaceAll later)
-  fRegexp->Substitute(trigger, "$1 ", "g");
-  
-  while (1) {
-    AliDebug(AliLog::kDebug, trigger.Data());
-    
-    TArrayI pos;
-    Int_t nMatches = fRegexp->Match(trigger, "", 0, 2, &pos);
-    
-    if (nMatches <= 0) break;
-    
-    TString token(trigger(pos[0], pos[1]-pos[0]+1));
-
-    TParameter<Int_t>* param = dynamic_cast<TParameter<Int_t> *>(fCashedTokens->FindObject(token));
-    if (!param) {
-      TInterpreter::EErrorCode error;
-      Int_t bit = gInterpreter->ProcessLine(Form("AliTriggerAnalysis::k%s;", token.Data()), &error);
-      
-      if (error > 0) AliFatal(Form("Trigger token %s unknown", token.Data()));
-      
-      param = new TParameter<Int_t>(token, bit);
-      fCashedTokens->Add(param);
-      AliDebug(AliLog::kDebug, "Added token");
+/// Evaluate if the given event fulfills a given trigger logic
+///
+/// \param event Pointer to the current event
+/// \param triggerAnalysis Pointer to the TriggerAnlysis class
+/// \param triggerLogic Describing trigger logic; e.g. "V0A && V0C && ZDCTime && !TPCHVdip"
+/// \param offline Offline analysis(?)
+///
+/// \return True if the given event matches the trigger logic
+Bool_t AliPhysicsSelection::EvaluateTriggerLogic(const AliVEvent* event,
+						 AliTriggerAnalysis* triggerAnalysis,
+						 const char* triggerLogic, Bool_t offline){
+  using namespace std;
+  // split the trigger logic on the individual trigger names and replace them
+  // with TFormula parameters
+  string trg_logic_orig(triggerLogic);
+  // Regex identifying the trigger (>=1; alphanumeric and _)
+  regex trg_re("\\w+");
```

Comment:
aha! Man I wish we'd have unit tests... Anyways, given that the cxx11 regex is out of the game, I reverted back to the TPRegex one with the old pattern